### DISCUSSION TITLE
Terrain causes uncontrollable opacity toggling of custom marker.

### DISCUSSION THREAD

**Discussion by sirflyingv** [maplibre/maplibre-gl-js#1782] [Q&A] [answered]
- amazon rgb terrain
- markers with custom html element
- when moving the view, maplibre changes style properties of markers and also toggles opacity 1/0.2 probably when terrain tile loads or something. 

I couldn't find any info in docs on this kind of relations between elements, can you help me? I suppose it have something to do with underground objects and their visualization, but there should be some options for that behavior.
 Here is video https://www.youtube.com/watch?v=6zqEOcTH2Pw , sorry for not optimal resolution.

**Comment by HarelM** [comment, marked as answer]
It's weird to see this in this pitch, but in theory, markers that are "behind" the terrain should be more transparent. But it might be that this is not working perfect.
I don't think this was documented anywhere, I saw this when testing and reviewing the terrain feature. Feel free to create a PR for docs update and or bug report.
